src/ncp/ncp_base.cpp

```diff
@@ -1671,10 +1827,21 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_MAC_SCAN_STATE>(void)
             VerifyOrExit(mCurScanChannel == kInvalidScanChannel, error = OT_ERROR_INVALID_STATE);
             VerifyOrExit(HasOnly1BitSet(mScanChannelMask), error = OT_ERROR_INVALID_ARGS);
 
-            scanChannel     = IndexOfMSB(mScanChannelMask);
-            mCurScanChannel = static_cast<int8_t>(scanChannel);
+            scanChannel = IndexOfMSB(mScanChannelMask);
 
             error = otLinkRawEnergyScan(mInstance, scanChannel, mScanPeriod, LinkRawEnergyScanDone);
+
+            if (error != OT_ERROR_NONE)
```

Comment:
The function `otLinkRawEnergyScan` may return the error `OT_ERROR_NOT_IMPLEMENTED`. In this case, there is no need to enqueue the EnergyScan command.

Do we need to call `otLinkRawIsTransmittingOrScanning()` before calling `otLinkRawEnergyScan ()`?

Reply:
Good point! Made changes to consider `OT_ERROR_NOT_IMPLEMENTED` error status. 

Regarding `otLinkRawIsTransmittingOrScanning(), we can call it but shouldn't Submac handle it? If the radio is busy with other activity and will error out anyways.